corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -781,15 +875,18 @@ function getEntry(question) {
         case Formplayer.Const.STRING:
             // Barcode uses text box for CloudCare so it's possible to still enter a barcode field
         case Formplayer.Const.BARCODE:
-            isNumeric = style === Formplayer.Const.NUMERIC;
-            if (isNumeric) {
-                entry = new PhoneEntry(question, {
-                    enableAutoUpdate: isPhoneMode,
+            options = {
+                enableAutoUpdate: isPhoneMode,
+                receiveStyle: (question.stylesContains(/receive-*/)) ? question.stylesContaining(/receive-*/)[0]: null,
+            }
+            if (question.stylesContains('address')) {
```

Comment:
TODO: Make `address` and the other regexes constants like `Formplayer.Const.NUMERIC)`

Reply:
thanks for noting, I was going to comment on that :)